chromium_src/third_party/blink/renderer/platform/weborigin/security_origin.cc

```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "third_party/blink/renderer/platform/wtf/text/string_builder.h"
+#include "third_party/blink/renderer/platform/wtf/text/wtf_string.h"
+
+namespace {
+
+const char* kIpfs[] = {"ipfs", ".ipfs.localhost"};
+const char* kIpns[] = {"ipns", ".ipns.localhost"};
+
+bool BuildRawIPFSIfApplicable(const String& host,
```

Comment:
The new string representation appears to be insecure (as opposed to `http://hash.ipfs.localhost`), so callers of `ToString()` could get confused. From quick searching there are `ServiceWorkerContainer::registerServiceWorker`, `IsInCrossOriginIframe` in password autofill agent or `ResourceRequestHead::SetHTTPOrigin`. 

I'd prefer to have one of the following 
 * add a comment explainig why the change is safe for us
 * do less invasive change, e.g. modify only `document.location`, not `SecurityOrigin`
 * investigate more detailed approach, e.g. adding `ipfs` to secure schemes. Also note that `.ipfs.localhost` plays well with eTLD+1 checking, while `ipfs://hash` does not

Reply:
`do less invasive change, e.g. modify only document.location, not SecurityOrigin`
done